### ISSUE TITLE
Allow type translation and more

### ISSUE THREAD

**Issue by siavashkavousi** [epoberezkin/ajv-i18n#54]
Hi,

I think there should some way to translate prop.type and things like that.

Thanks.

**Comment by epoberezkin** [comment]
@siavashkavousi can you please clarify.

**Comment by siavashkavousi** [comment]
if we have the following message in en and my locale (fa)
en: should be {{=e.params.type}}
fa: 'فیلد باید {{=e.params.type}} باشد'

then I got 
‍`فیلد باید number باشد`
like in english 
`should be number`

as you can see `number` can be localized as well :)

**Comment by epoberezkin** [comment]
Makes sense for type names. So it would require a map of type names for each language (and by default english will be used, if the map is not provided).

**Comment by epoberezkin** [comment]
What do you mean by "and more"?

**Comment by siavashkavousi** [comment]
I didn't quite integrate it in my application, but one more thing which I think can be localized is property names. 
I mean by passing title instead of property name, messages looks better (in my application) which I think is not general.

**Comment by epoberezkin** [comment]
That's definitely more complex, it has to be somehow passed via ajv. Maybe ajv-errors can be extended to support i18n.

**Comment by adefrutoscasado** [comment]
Same problem here, is there any workaround? Being able to pass a translate function would be nice

**Comment by FedeTommi** [comment]
I have the same problem too.

I had a look at the jtd.js and index.js and it seems like the problem is solved for the languages fi, nb, nl and pl.
Here a proposal for the languages German and Italian:

de: '{{#def.t}}muss {{? t == "number" }}eine Zahl{{?? t == "integer"}}eine ganze Zahl{{?? t == "string"}}ein Text{{?? t == "boolean"}} boolescher Wert{{??}}{{=t}}{{?}}{{=n}} sein',

it: '{{#def.t}}deve essere {{? t == "number" }}un numero{{?? t == "integer"}}un numero intero{{?? t == "string"}}un testo{{?? t == "boolean"}} booleano{{??}}{{=t}}{{?}}{{=n}}',
 
 Would it be possible to add these?
